src/scene/vcPolyModelNode.cpp

```diff
@@ -46,8 +46,7 @@ vcPolyModelNode::vcPolyModelNode(vdkProject *pProject, vdkProjectNode *pNode, vc
     pLoadInfo->pNode = this;
     pLoadInfo->pProgramState = pProgramState;
 
-    // Queue for load
-    udWorkerPool_AddTask(pProgramState->pWorkerPool, vcPolyModelNode_LoadModel, pLoadInfo, true);
+    vcPolyModelNode_LoadModel(pLoadInfo);
```

Comment:
I think this leaks memory now - the final parameter was `true` which is to `clearMemory` which means it will call `udFree` on the user data, `pLoadInfo`. You could have alternatively changed the call to `udWorkerPool_AddTask` to use the `postFunction` instead of the `func` parameter for `vcPolyModelNode_LoadModel`, this would keep the work done via the worker pool, but will perform the work on the main thread instead.

@OsiJr Is this a case where using the worker pool makes sense?

Reply:
The memory is freed after the post- the model creation should definately be done in the post function. Some VSM files take >10s to load so can't happen on the main thread.

Reply:
I see. It looks like the solution will be to break up `vcPolyModelNode_LoadModel` into worker-thread-work and main-thread-work, using both `func` and `postFunction`. Thoughts?

Reply:
Righto I found the issue. See latest